### DISCUSSION TITLE
how do you use a TTK styles definition file?

### DISCUSSION THREAD

**Discussion by lildinti** [alejandroautalan/pygubu-designer#131] [General]
Hi Everyone. 
Does anyone know how to use a ttk styles definition file? There's a checkbox in the code generation screen of the designer though. Any idea how it works. 

Cheers
lildinti

PS:  I know how to set style in cod, juat interested in the file and how to use it!

**Comment by alejandroautalan** [comment]
Hello @lildinti, the usage is explained [here](https://github.com/alejandroautalan/pygubu-designer/wiki/Ttk-Styles-Support).
The advantage of the style file is that:

> Pygubu Designer will monitor this file and load the changes automatically.

So, you edit the styles (in your preferred editor) and pygubu-designer will show in real-time the changes.

If you have any questions let me know.

Regards
Alejandro A.

**Comment by lildinti** [comment]
Thanks alejandro,

It seems i need to add these lines before to the code generated:

```
  self.style = ttk.Style()
  self.style.theme_use('default')
```
before the 
```
   self.setup_ttk_styles()
```
Is that correct? Otherwise, seems the custom style isn't applied....
Cheers,
Claudio

**Reply by alejandroautalan** [reply]
That's weird, shouldn't be necessary. If you can make a minimal example to review, it would be great.
What operating system are you using?

**Comment by lildinti** [comment]
I'm using MacOs 13.6 sierra high. Havent tried big sur yet.
Here's my styles file content:
```
style.configure("Header.TButton",
    background="#388E3C",
    foreground="white")
# configure button highlight colours
style.map('Header.TButton', 
    background=[('active','#CDDC39')],
    foreground=[('active','black')])
```
The only thing i can think of mucking the code is the style.map command. What do you think? also on the same topic, i saw theres a reference to the optionsdb. is it possible to do any configuration using the optionsdb? (for example the list component of a combobox).
Cheers
Claudio
PS: sorry about being a pain in the arse!

**Reply by alejandroautalan** [reply]
Hello. I think I know what is happening here.

In MacOS the default active theme is 'aqua'. This theme does not allow some option modifications as stated [here](https://tkdocs.com/tutorial/styles.html):

> You'll sometimes try to change an option that is supposed to exist according to element options, but it will have no effect. For example, you can't modify the background color of a button in the aqua theme used by macOS. While there are valid reasons for these cases, it's not easy to discover them, which can make experimenting frustrating at times. (tkdocs.com)

So, you will need to change to a theme that allow your changes.

> i saw theres a reference to the optionsdb. is it possible to do any configuration using the optionsdb? (for example the list component of a combobox).

At the moment with the UI file can't be used because the options must be setup before widget creation.  I need to send the function **setup_ttk_styles** as a parameter of the builder to be called after the first toplevel is created, and before any child widget creation.

Regards
Alejandro A.

**Comment by lildinti** [comment]
Awesome... yes unfortunately Macs are a beast of their own! 
Anyway, for all the mac users the trick is to include those lines in their code!. 
Cheers, mate
